etc/zfs/Makefile.am

```diff
@@ -13,3 +14,5 @@ pkgsysconf_SCRIPTS = \
 	zfs-functions
 
 SUBSTFILES += $(pkgsysconf_SCRIPTS)
+
+SHELLCHECK_SHELL = dash
```

Comment:
Thankfully, Busybox ash is pretty much equivalent to dash feature-wise, considering.
I've trimmed down the errors we get with sh down to just `local` variables, which I'm *definitely* not qualified to remove – added appropriate comment to the SHELLCHECK_SHELL lines. I find it preferable to muting SC2039, since SC2039 might shadow something Almquist shells don't support, either.

Hm, I tested on buster, which has 0.5.0 – fixed and re-ran on sid, too. Fixed.

Reply:
I verified the `shellcheck` and `checkbashisms` targets in the updated PR on Fedora 33 and didn't encounter any problems.  The shell cleanups themselves all look reasonable to me but I didn't try and manually test them all.